libcontainer/cgroups/fs2/fs2.go

```diff
@@ -158,7 +158,7 @@ func (m *manager) Freeze(state configs.FreezerState) error {
 }
 
 func (m *manager) Destroy() error {
-	if err := os.Remove(m.dirPath); err != nil && !os.IsNotExist(err) {
+	if err := os.RemoveAll(m.dirPath); err != nil && !os.IsNotExist(err) {
```

Comment:
1. Theoretically speaking, we should only remove the directories, as cgroup file removes is not supported and will result in some error (possibly `EPERM`). I am not sure how RemoveAll() works, but ...

2. os.RemoveAll() doc says:

> It removes everything it can but returns the first error it encounters. If the path does not exist, RemoveAll returns nil (no error).

so 
* it should not return `IsNotExist()`, and thus the check for it is not needed
* it might return some other error, which we'll return.

All of this makes me think we have to write a removal function based on `filepath.Walk` or `ioutil.ReadDir`, which will only remove directories.

Reply:
But maybe I'm wrong and this works fine as it is. This is why we need a test case or two.

Reply:
> * it should not return `IsNotExist()`, and thus the check for it is not needed

Yes, removed now.

Reply:
As Kir said, `filepath.Walk` is the right way.

Reply:
> As Kir said, `filepath.Walk` is the right way.

Yes, I looked the code of `os.RemoveAll`, it does removes everything it can. But ignore the `EPERM` error.
```
if err != syscall.EISDIR && err != syscall.EPERM && err != syscall.EACCES {
```
So, it does not worth to use `os.RemoveAll`.
Changed to `ioutil.ReadDir`.